### PR TITLE
Increase tests coverage for the Core folder - NotificationObject

### PR DESCRIPTION
**Pull request by byorda-glb**
### Purpose

This PR adds the required tests to increase coverage on the NotificationObject class to 100%.

### Declarations

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] All tests pass using the self-service CI.
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

Aaron Tang (QilongTang)
Michael Kirschner (mjkkirschner)

### FYIs

Alfredo Pozo (alfredo-pozo)

### CODE REVIEW

**Comment by byorda-glb** [comment]
Here is the DotCover coverage report:

image

All the tests passed in the SelfServe CI job, execution 32.

File: test/DynamoCoreTests/Core/NotificationObjectTests.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        /// Derivate class from NotificationObject, designed to help in its testing
+        /// </summary>
+        private class NotificationObjectTestingSubclass : NotificationObject
+        {
+            private bool _testProperty;
+
+            /// <summary>
+            /// Used to test the RaisePopertyChanged method that receives a single property
+            /// </summary>
+            public bool TestPropertySingle
```
Comment (QilongTang) [inline test/DynamoCoreTests/Core/NotificationObjectTests.cs:57]: Would you use `internal` for these properties?

Comment (byorda-glb) [reply test/DynamoCoreTests/Core/NotificationObjectTests.cs]: Done, I also made the constructor of the utility class and the flag internal.

**Comment by QilongTang** [comment]
LGTM

**Pull request merged**